### PR TITLE
Intorduce shim for g_autofd and fix memory leak

### PR DESCRIPTION
**Pull request by jsparber**

### CODE REVIEW

File: libportal/portal-private.h [hunk 1]
```diff
@@ -71,3 +74,16 @@ const char * portal_get_bus_name (void);
...
+static inline void
+shim_g_clear_fd_ignore_error (int *fdp)
+{
+  int sv_errno = errno;
+  int fd = g_steal_fd (fdp);
+  if (fd >= 0)
+    g_close (fd, NULL);
+  errno = sv_errno;
+}
+#define g_autofd2 _GLIB_CLEANUP(shim_g_clear_fd_ignore_error)
```
Comment (swick) [inline libportal/portal-private.h:88]: `g_autofd2`?

Comment (jsparber) [reply libportal/portal-private.h:88]: forgot to remove it :facepalm:

File: libportal/portal-private.h [hunk 2]
```diff
@@ -71,3 +74,44 @@ const char * portal_get_bus_name (void);
...
+       * is a programming error, checked for by g_close(). */
+    }
+
+  G_GNUC_END_IGNORE_DEPRECATIONS
+
+  errno = errsv;
+}
+
+#define g_autofd __attribute__((cleanup(_g_clear_fd_ignore_error)))
+#endif
```
Comment (GeorgesStavracas) [inline libportal/portal-private.h:117]: Hmmm I'd rather put this in a `glib-backports.h` file, and add an `#include "glib-backports.h` here

Comment (jsparber) [reply libportal/portal-private.h]: done

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**